### PR TITLE
feat: Switch custom webpack build step to ember-auto-import

### LINKED ISSUE #145: Use ember-auto-import for packaging?
Should we be using [ember-auto-import](https://github.com/ef4/ember-auto-import) for packaging dependencies? Any advantages/disadvantages compared to our current setup?

The obvious advantage is that it's a shared community solution and we'd no longer need to maintain our own packaging setup. Also it should make for a clean transition to future ember versions where that functionality is available out-of-the-box.

Thoughts? Especially @dfreeman 😄

### PR DESCRIPTION
**Pull request by jasonmit**
fixes #145, #152

Side note, would much appreciate a beta or an rc tag once this is merged.  Will enable me to get off my fork that hacked together engine support.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,45 +1,17 @@
 'use strict';
 
-const apolloClientDefaultPackages = [
-  'apollo-cache',
-  'apollo-cache-inmemory',
-  'apollo-client',
-  'apollo-link',
-  'apollo-link-context',
-  'apollo-link-http',
-  'graphql',
```
Comment (jasonmit) [inline index.js]: graphql-js is the problematic dependency.  https://github.com/graphql/graphql-js/issues/1272

**Comment by jasonmit** [comment]
@dfreeman & @bgentry I resolved the bundling issues with `graphql-js`

Let me know if there is anything else needed to get this merged.

Thanks again!

File: index.js [hunk 2]
```diff
@@ -1,45 +1,24 @@
...
-
-    return new WebpackDependencyPlugin({
-      outputName: 'apollo-client',
-      expose: [...includedPackages, ...userPackages],
-    });
+  options: {
+    autoImport: {
+      webpack: {
+        module: {
+          rules: [
```
Comment (jasonmit) [inline index.js]: Only needed for `0.13.1` & `0.13.2`.  This issue does not exist in `0.13.0` and `>= 14.0.0-rc.1`.  Verified configuration is safe for all versions of `graphql-js`.

File: index.js [hunk 3]
```diff
@@ -1,45 +1,24 @@
...
+            /* fixes issue with graphql-js's mjs entry */
+            /* see: https://github.com/graphql/graphql-js/issues/1272#issuecomment-393903706 */
+            {
+              test: /\.mjs$/,
+              include: /node_modules/,
```
Comment (jasonmit) [inline index.js]: I think I can scope this further to just `graphql-js`

**Comment by bgentry** [comment]
Thanks @jasonmit, this looks great! Since it is a breaking change, prior to merging and releasing it I want to finish up #156 to deprecate the old mixin import paths. Then we can release a v2 beta with this new packaging setup and the old mixin imports removed entirely.

**Comment by jasonmit** [comment]
@bgentry friendly ping.  Let me know if there is anything I can do.

**Comment by bgentry** [comment]
Thanks @jasonmit! The blocker remains that I want to deprecate the old import paths in a release before the major upgrade. So if you’re able to help with that it would be 🙏🏻

I guess another option would be to make a new branch for 1.x, then merge this and release a 2.0 beta that doesn’t yet include the deprecations. So I could do that pretty soon if nobody is able to help with #156

**Comment by jasonmit** [comment]
@bgentry https://github.com/bgentry/ember-apollo-client/pull/167 thanks!

**Comment by josemarluedke** [comment]
Any plans on getting this merged/released soon?

**Comment by bgentry** [comment]
One thing that appears to have been lost with this PR is that the readme no longer explains how to import additional packages. Since folks are likely coming from 1.x where this is a bundled feature, maybe we should explain what they get out-of-the-box and show an example or at least link to ember-auto-import for how to get additional Apollo-related packages loaded?

Can anybody PR that small doc fix today? Then I'll publish a v2.x beta.

Thanks so much for this PR @jasonmit 👏 sorry for how long this took to ship 😞

**Pull request merged**